compiler/src/middle_end/matchcomp.re

```diff
@@ -502,7 +496,7 @@ module MatchTreeCompiler = {
                 cmp_id_name,
                 Comp.prim2(
                   ~allocation_type=StackAllocated(WasmI32),
-                  Eq,
+                  Is,
```

Comment:
Calling `Pervasives.(==)` was unnecessary here (and prevented pattern matching on variants to happen in runtime mode). This actually will speed up pattern matching by a non-trivial amount.